content/projects/python-specific/file-io/_index.md

```diff
@@ -3,12 +3,90 @@ _db_id: 695
 content_type: project
 flavours:
 - python
-ready: false
+ready: true
 submission_type: repo
 tags:
 - FileIO
 title: interacting with files
 ---
 
-TODO: take a look at this: content/projects/nodejs/file-io/_index.md 
-Make a projet spec that works
\ No newline at end of file
+You are required to create a back-end service that will help capture basic information about prospective students who come to inquire here at Umuzi. In this project you'll just be storing and retrieving information from plain old json files.
+
+## Instructions
+
+1. Create a class called Visitor. Instances of this class should have the following properties:
+
+- full name
+- age
+- date of visit
+- time of visit
+- comments
+- name of the person who assisted the visitor
+
+2. Create a function called `save` that saves the visitor's data to a JSON file. The file name should be named like this `visitor_{their_full_name}.json`.
+
+```
+alice.save()   # results in visitor_alice_cooper.json
+bob.save()     # results in visitor_bob_marley.json
+charlie.save() # results in visitor_charley_sheen.json
+```
+
+Notice that the full name used in the file is all lower-case and spaces are replaced by underscores.
+
+3. Create a function called `load` that takes in a name and then grabs a Visitor object from file. It should simply `print` the visitor.
+
+eg:
+
+```
+load("Alice Cooper")
+# prints out all of Alice's goodies
+
+
+load("Bob Marley")
+# Same deal for good ol Bob
+```
+
+## Resources
+
+- [Working With Files in Python](https://realpython.com/working-with-files-in-python/)
+- [JSON](https://www.w3schools.com/js/js_json_intro.asp): Make sure you understand everything up to the end of "JSON Arrays"
+- [Python JSON package](https://www.w3schools.com/python/python_json.asp)
+
+## Up for a challenge?
+
+Here are some upgrades you can add to your project if you are up for it.
+
+4. Update your `load` function so that it returns an instance of `Visitor` instead of just `print`ging it. You'll need to learn a little bit about Synchronous versus Asynchronous code to get this one right :)
```

Comment:
This can be removed.

Reply:
The instructions should say that the load function should return a visitor instance. So having extra instructions thatsay the same thing as the main instructions wouldn't make sense.